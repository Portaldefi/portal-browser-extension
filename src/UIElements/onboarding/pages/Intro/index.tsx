import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Header, Image, Button, Grid, Segment, Icon } from 'semantic-ui-react';

export default () => {
  const navigate = useNavigate();

  const handleContinue = useCallback(() => {
    navigate('/select-action');
  }, []);
  const handleBack = useCallback(() => {
    navigate('/welcome')
  }, []);

  return (
    <Segment className='board'>
      <Grid>
        <Grid.Row centered>
          <Header size='medium' className='heading'>Our Values</Header>
        </Grid.Row>
        <Grid.Row centered columns={2}>
          <Grid.Column>
            <Grid.Row centered>
              <Header size='medium' className='heading'>Censorship-resistant</Header>
              <Header as='p' size='small' color='grey' className='description'>Using Fabric, we enable web scale, serverless infrastructure with strong privacy and security guarantees.</Header>
              <Image src='/images/onboarding/black-glasses.png' />
            </Grid.Row>
          </Grid.Column>

          <Grid.Column>
            <Grid.Row centered>
              <Header size='medium' className='heading'>Freedom</Header>
              <Header as='p' size='small' color='grey' className='description'>Fabric, the underlying protocol Portal uses extends Bitcoinâ€™s functionality today without needing any BIPs.</Header>
              <Image src='/images/onboarding/portal.png' />
            </Grid.Row>
            
          </Grid.Column>

        </Grid.Row>
        <Grid.Row centered columns={2}>
          <Grid.Column>
            <Button className='blank-button' onClick={handleBack}><Icon name='angle left' />Back</Button>
          </Grid.Column>
          <Grid.Column>
            <Button className='primary-button' onClick={handleContinue}>Continue</Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Segment>
  )
}