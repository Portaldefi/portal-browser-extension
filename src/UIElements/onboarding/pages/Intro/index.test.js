import Intro from './index';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Button } from 'semantic-ui-react';

describe('IntroComponent', () => {
  it('should work', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <Intro />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  it('buttons', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <Intro />
        </BrowserRouter>
      );
    });

    const buttons = tree.root.findAllByType(Button);
    act(() => buttons[0].props.onClick());
    act(() => buttons[1].props.onClick());
    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});