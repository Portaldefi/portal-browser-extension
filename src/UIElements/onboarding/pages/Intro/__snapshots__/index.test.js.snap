// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroComponent buttons 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        Our Values
      </div>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <div
          className="centered row"
        >
          <div
            className="ui medium header heading"
          >
            Censorship-resistant
          </div>
          <p
            className="ui grey small header description"
          >
            Using Fabric, we enable web scale, serverless infrastructure with strong privacy and security guarantees.
          </p>
          <img
            className="ui image"
            src="/images/onboarding/black-glasses.png"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="centered row"
        >
          <div
            className="ui medium header heading"
          >
            Freedom
          </div>
          <p
            className="ui grey small header description"
          >
            Fabric, the underlying protocol Portal uses extends Bitcoin’s functionality today without needing any BIPs.
          </p>
          <img
            className="ui image"
            src="/images/onboarding/portal.png"
          />
        </div>
      </div>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <button
          className="ui button blank-button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="angle left icon"
            onClick={[Function]}
          />
          Back
        </button>
      </div>
      <div
        className="column"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Continue
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`IntroComponent should work 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        Our Values
      </div>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <div
          className="centered row"
        >
          <div
            className="ui medium header heading"
          >
            Censorship-resistant
          </div>
          <p
            className="ui grey small header description"
          >
            Using Fabric, we enable web scale, serverless infrastructure with strong privacy and security guarantees.
          </p>
          <img
            className="ui image"
            src="/images/onboarding/black-glasses.png"
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="centered row"
        >
          <div
            className="ui medium header heading"
          >
            Freedom
          </div>
          <p
            className="ui grey small header description"
          >
            Fabric, the underlying protocol Portal uses extends Bitcoin’s functionality today without needing any BIPs.
          </p>
          <img
            className="ui image"
            src="/images/onboarding/portal.png"
          />
        </div>
      </div>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column"
      >
        <button
          className="ui button blank-button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="angle left icon"
            onClick={[Function]}
          />
          Back
        </button>
      </div>
      <div
        className="column"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Continue
        </button>
      </div>
    </div>
  </div>
</div>
`;
